### PR TITLE
Feature/docker update

### PR DESCRIPTION
**Pull request by lebdron**
## What is this pull request?
Update docker images, and scripts for docker.
   
- Remove misleading build instructions, which were used in previous versions.
- Remove old configuration files and scripts
- Add dependencies to docker image to allow faster build in future updates

## How to use
See `scripts/README.md`, or simply run `scripts/run-iroha-dev.sh`

### CODE REVIEW

**Comment by neewy** [comment]
Minor question:
if I run `sh compose-run.sh` I get following:
`ERROR: .IOError: [Errno 2] No such file or directory: u'././docker/docker-compose.yml'`
If I run it from Iroha folder it runs fine, like `sh scripts/compose-run.sh`

**Comment by tkyonezu** [comment]
I think the reason become with below environment variables.

  SOURCE=$(dirname $(dirname “$0”))

But, I think ${IROHA_HOME} is more useful rather than this.

  SOURCE=${IROHA_HOME}

And, I think Makefile is more useful for building docker and so on, which I
wrote it previously.

File: scripts/README.md [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+Build on docker container for develop environment `hyperledger/iroha-docker-develop` so you need to build it:
+
+```bash
+docker build -t hyperledger/iroha-docker-develop /path/to/iroha/docker/develop
+```
+
+# How to run
+
+```bash
+/path/to/iroha/scripts/run-iroha.dev.sh
```
Comment (muratovv) [inline scripts/README.md]: `sh /path/to/iroha/scripts/run-iroha.dev.sh` or `./path/to/iroha/scripts/run-iroha.dev.sh`

File: scripts/README.md [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+```
+
+You will be attached to interactive environment for development and testing
+
```
Comment (muratovv) [inline scripts/README.md]: Add build and test commands:
```bash
cmake -H. -Bbuild; cmake --build build -- -j`npoc`;
```
And tests with
```bash
cmake --build build --target test;
```

**Pull request merged**